File: python/83_remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if head is None:
            return None
        cur = head
        Max = head

        while cur:
            if cur.val != Max.val:
                Max.next = cur
                Max = cur

            cur = cur.next

        Max.next = None
        return head

File: go/83_remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	curNode := head
	for curNode != nil && curNode.Next != nil {
		for curNode.Next != nil && curNode.Val == curNode.Next.Val {
			curNode.Next = curNode.Next.Next
		}
		curNode = cur.Next
	}
	return head
}